napari/layers/points/_points_key_bindings.py

```diff
@@ -58,7 +60,61 @@ def paste(layer: Points):
     trans._("Select all points in the current view slice."),
 )
 def select_all(layer: Points):
-    layer.selected_data = set(layer._indices_view[: len(layer._view_data)])
+    new_selected = set(layer._indices_view[: len(layer._view_data)])
+
+    # If all visible points are already selected, deselect the visible points
+    if new_selected & layer.selected_data == new_selected:
+        layer.selected_data = layer.selected_data - new_selected
+        show_info(
+            trans._(
+                "Deselected all points in this slice,"
+                " use Shift-A to deselect all points on the layer."
+                f" ({len(layer.selected_data)} selected)"
+            )
+        )
```

Comment:
I find this logic a bit iffy... Personally, I would only deselect the whole slice if the sets are equal. It's weird (imho) to *only* select points not in a slice? I can't think of any software that does this this way?

Reply:
So it means:
If all points on the current slice are selected, deselect all points of the current slice.
Otherwise, select all points of the current slice.
Leave other slices alone.

Example:

slice 0: points 1 and 2
slice 1: point 3
slice 2: point 4 and 5

I selected slice 0 already:
   layer.selected_data = {1, 2}
I press A on slice 2:
   new_selected = {4, 5}
Therefore the intersection is
    layer.selected_data & new_selected = {} != {4, 5}
And the else is triggered assigning the union of both sets:
   layer.selected_data = layer.selected_data | new_selected = {1, 2, 4, 5}

Now when I press A again on layer 2, the intersection is  
    layer.selected_data & new_selected = {4, 5} == {4, 5}
Triggering the if, leading to new_selected to be removed from the set:
    layer.selected_data = layer.selected_data - new_selected = {1, 2}

Reply:
Got it! Yes I had indeed misunderstood and thought that the selection was being inverted.